core/server/api_container/server/startosis_engine/startosis_warning/warning.go

```diff
@@ -0,0 +1,38 @@
+package startosis_warning
+
+import (
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"sync"
+)
+
+var warningMessageSet *sync.Map
+
+// Initialize This method is responsible for creating warningMessageSet; if the warning set already exists it will override
+// that with an empty warningMessageSet. This method is intended to be used during initialization process.
+func Initialize() {
+	warningMessageSet = new(sync.Map)
+}
+
+func PrintOnceAtTheEndOfExecutionf(message string, args ...interface{}) {
+	if warningMessageSet != nil {
+		formattedMessage := fmt.Sprintf(message, args...)
+		_, stored := warningMessageSet.LoadOrStore(formattedMessage, true)
+		if !stored {
+			logrus.Warnf("Error occurred while adding warning to the set with message: %v", formattedMessage)
+		}
+	}
```

Comment:
what if the warning message set is nil? do we fail silently? seems off

throwing it out there - can we initialize it the first time its called if it isn't initialized? the current way seems a bit concerning. our code path continues to work if I go and remove the `startosis_warning.Initialize()`

Reply:
I was debating between making it silent failures or not - because it's warnings and don't want to discomfort users with it's failure; but yeah in the hindsight it's fine because it can be caught while development if it the error is verbose. Let me wrap it in `once` to actually avoid it.